### PR TITLE
soft98.ir

### PR DESCRIPTION
**Pull request by MasterKia**
Address: https://github.com/AdguardTeam/AdguardFilters/issues/126252

Related: https://github.com/uBlockOrigin/uAssets/pull/14278

I'll send pull request everyday to update the filters.

cc @zloyden @Yuki2718

### CODE REVIEW

**Comment by MasterKia** [comment]
I could convert uBO to AdGuard syntax if that's really necessary.

**Comment by MasterKia** [comment]
I get ` The 'MasterKia/AdguardFilters' repository doesn't contain the 'EnglishFilter/sections/foreign.txt' path in 'MasterKia-patch2/EnglishFilter/sections/foreign.txt'. ` when I try to edit the file.

**Review by Yuki2718 — COMMENTED** [review]
Just a general comment

```
soft98.ir##+js(acis, document.oncontextmenu)
soft98.ir##[src*="file.soft98.ir"]:upward(3):remove()
```
=>
```
soft98.ir#%#//scriptlet("abort-current-inline-script", "document.oncontextmenu")
soft98.ir#$?#[src*="file.soft98.ir"]:upward(3)  { remove: true; }
```
However, a rule for `document.oncontextmenu` should be in Annoyances if it's meant to tackle right click issue.

`||file.soft98.ir^$all,redirect=nooptext` => `||file.soft98.ir^$script,xmlhttprequest,redirect=nooptext` (should be this way for uBO too)

`soft98.ir##body > :nth-of-type(1):before:style(content: "در صورت دیدن هرگونه مشکل در سایت و یا پیغامی برای غیرفعال کردن افزونه حذف تبلیغات، کافیست به تنظیمات افزونه مسدودسازتان بروید و لیست AdGuard Base و PersianBlocker را بروز کنید." !important; color: #ff0000 !important; visibility: visible; font-size: large !important;)`
=>
`soft98.ir#$#body > :first-of-type:before { content: "در صورت دیدن هرگونه مشکل در سایت و یا پیغامی برای غیرفعال کردن افزونه حذف تبلیغات، کافیست به تنظیمات افزونه مسدودسازتان بروید و لیست AdGuard Base و PersianBlocker را بروز کنید." !important; color: #ff0000 !important; visibility: visible; font-size: large !important; }`

`@@||soft98.ir^$ghide` => `@@||soft98.ir^$generichide`

**Comment by MasterKia** [comment]
@Yuki2718 Doesn't AdGuard translate these uBo syntax to AG compatible syntax internally?

**Comment by MasterKia** [comment]
> However, a rule for document.oncontextmenu should be in Annoyances if it's meant to tackle right click issue.

I'll drop this rule, no longer ~needed~ effective.

> `||file.soft98.ir^$script,xmlhttprequest,redirect=nooptext` (should be this way for uBO too)

Doesn't work. It's supposed to remove the banner at the top [like this](https://user-images.githubusercontent.com/17685483/183246869-903e2b58-1640-4bf5-8a54-6d4820669f2b.png), but your suggest filter doesn't do that..

**Comment by Yuki2718** [comment]
> Doesn't work. It's supposed to remove the banner at the top [like this](https://user-images.githubusercontent.com/17685483/183246869-903e2b58-1640-4bf5-8a54-6d4820669f2b.png), but your suggest filter doesn't do that on uBO.

Then use according request type and corresponding redirect resource, $all is not very good for redirect.

**Comment by MasterKia** [comment]
> Then use corresponding redirect resource

1x1.gif (and similar) => results in a big white square
noopframe => doesn't work
nooptext and click2load.html work.

**Comment by MasterKia** [comment]
@Yuki2718 I can't edit the file.
https://github.com/AdguardTeam/AdguardFilters/pull/126425#issuecomment-1207194512

**Comment by Yuki2718** [comment]
> I can't edit the file.?

What? You can edit whatever file in you branch, right?

**Comment by MasterKia** [comment]
@Yuki2718 Please check if the syntax is acceptable.
Also the could you test the `:before:style()` filter on Chromium? For me it only works on Firefox.

File: EnglishFilter/sections/foreign.txt [hunk 1]
```diff
@@ -6459,11 +6459,36 @@ zoomit.ir#$#body[style^="padding-top:"] { padding-top: 0!important; }
 zoomit.ir#$#body > .hotMenu[style="top: auto;"] { top: 0!important; }
 zoomit.ir#$#body > nav.is-sticky[style="top: 130px;"] { top: 0!important; }
 zoomit.ir#%#document.cookie = "BnShowNum=4; path=/;";
-soft98.ir#$#[id^="sidebar"][id*="stick"] > section { opacity: 0 !important; pointer-events: none !important; }
+soft98.ir#$#header[id]:has(.row) { remove: true; }
```
Comment (zloyden) [inline EnglishFilter/sections/foreign.txt:6462]: Use syntax `example.com#$?#.ad { remove: true; }` for such rules.
Btw, they cause adb detection.

These rules work fine on my end. A directive is needed to limit platforms.
```
!#if (!adguard_app_windows && !adguard_app_mac && !adguard_app_android)
||soft98.ir/engine/modules/antibot/antibot.php
||soft98.ir/engine/ajax/controller.php
||file.soft98.ir^$image,redirect=nooptext
||cdn.hostdl.com^$image,redirect=nooptext
||linkdoni.soft98.ir^
||iranicard.ir^$domain=soft98.ir
||asiatech.ir^$domain=soft98.ir
||sysban.ir^$domain=soft98.ir
||smostafa.ir^$domain=soft98.ir
soft98.ir#$#.show.fade.tooltip { display: none !important; }
!#endif
```

**Comment by zloyden** [comment]
> Doesn't AdGuard translate these uBo syntax to AG compatible syntax internally anyway?

Our compiler does both, but when you add rules to our filters, they must be written using [our syntax](https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters).

**Comment by MasterKia** [comment]
```
##header[id]:has(.row):remove() 
=>
#$?#header[id]:has(.row) {remove: true;}

##[src*="file.soft98.ir"]:upward(3):remove()
=>
#$?#[src*="file.soft98.ir"]:upward(3) {remove: true;}

^$ghide
=>
^$generichide

##+js(acis, document.oncontextmenu)
=>
#%#//scriptlet("abort-current-inline-script", "document.oncontextmenu")
```

**Comment by MasterKia** [comment]
@zloyden I tried to be compatible.
Please check.

File: EnglishFilter/sections/foreign.txt (hunk 1 above)
Comment (MasterKia) [reply EnglishFilter/sections/foreign.txt]: > they cause adb detection.

This site changes fast, if AdGuard merge filters slowly they'll become useless.

**Review by zloyden — COMMENTED** [review]
Added some of your rules directly https://github.com/AdguardTeam/AdguardFilters/commit/1ea38d963da3b5dfce50862790ad4400d0795b4f, I can't edit your branch.

**Comment by MasterKia** [comment]
The latest rules are:
```
soft98.ir##+js(rc, c6e1y2, *, stay)
soft98.ir##[src*="file.soft98.ir"]:style(color: #437593 !important; background-color: #437593 !important; border: #437593 !important;)
soft98.ir##[src*="cdn.hostdl.com"]:style(color: white !important; background-color: white !important; border: white !important;)
soft98.ir##:xpath(/html/body/div[*]/div/div[*]/div/div[1]):not(:has([role])):remove()
soft98.ir##[href*="linkdoni.soft98.ir"]:remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/div/section[*]/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/div):has(img[src*="cdn.hostdl.com"]):style(color: white !important;)
soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/div):remove()
soft98.ir##:xpath(/html/body/div[*]/div/div/div/section):remove()
soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/span):remove()
soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/div[*]):not(:has(img[src*="cdn.soft98.ir"])):not(:has-text(/)):not(:has-text(-)):not(:has(a[href*="/dl"])):remove()
soft98.ir##[href*="SMostafa.ir"]:remove()
||soft98.ir/engine/modules/antibot/antibot.php^$all
||soft98.ir/engine/ajax/controller.php^$all
||file.soft98.ir^$image,redirect=click2load.html
||cdn.hostdl.com^$image,redirect=click2load.html
||linkdoni.soft98.ir^$all
||iranicard.ir^$all,domain=soft98.ir
||iranicard.ir^$removeparam=/^utm_/
||asiatech.ir^$all,domain=soft98.ir
||asiatech.ir^$removeparam=/^utm_/
||cdn.hostdl.com/goto/soft98^$all
||sysban.ir^$all,domain=soft98.ir
||smostafa.ir^$all,domain=soft98.ir
@@||soft98.ir^$ghide
```

As you can see here, I'm constantly sending PRs and they get merged quickly:
https://github.com/uBlockOrigin/uAssets/pulls?q=is%3Apr+is%3Aclosed+author%3AMasterKia

**Comment by zloyden** [comment]
Not all of them are nedeed.
About using our syntax.
```
soft98.ir##+js(rc, c6e1y2, *, stay) - soft98.ir#%#//scripltet('remove-class', 'c6e1y2')
soft98.ir##[src*="cdn.hostdl.com"]:style(color: white !important; background-color: white !important; border: white !important;) - soft98.ir#$#[src*="cdn.hostdl.com"] { color: white !important; background-color: white !important; border: white !important; }
soft98.ir##[href*="linkdoni.soft98.ir"]:remove() - soft98.ir#$?#[href*="linkdoni.soft98.ir"] { remove: true; }
@@||soft98.ir^$ghide - @@||soft98.ir^$generichide
```
Why do you need a modifier `$all` - `||linkdoni.soft98.ir^$all`?

**Comment by MasterKia** [comment]
> Why do you need a modifier `$all` - `||linkdoni.soft98.ir^$all`?

I don't want them to get tricky with popups or redirects.

**Comment by zloyden** [comment]
I got it, but don't see anyu popups on my end.

**Comment by MasterKia** [comment]
Disable AdGuard, refresh the page and click on the Amazon banner in the header => it's a popup.

**Comment by Rusenche** [comment]
> The latest rules are:
> 
> ```
> soft98.ir##+js(rc, c6e1y2, *, stay)
> soft98.ir##[src*="file.soft98.ir"]:style(color: #437593 !important; background-color: #437593 !important; border: #437593 !important;)
> soft98.ir##[src*="cdn.hostdl.com"]:style(color: white !important; background-color: white !important; border: white !important;)
> soft98.ir##:xpath(/html/body/div[*]/div/div[*]/div/div[1]):not(:has([role])):remove()
> soft98.ir##[href*="linkdoni.soft98.ir"]:remove()
> soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/header):remove()
> soft98.ir##:xpath(/html/body/div[*]/div/div/section[*]/header):remove()
> soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/div):has(img[src*="cdn.hostdl.com"]):style(color: white !important;)
> soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/header):remove()
> soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/div):remove()
> soft98.ir##:xpath(/html/body/div[*]/div/div/div/section):remove()
> soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/span):remove()
> soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/div[*]):not(:has(img[src*="cdn.soft98.ir"])):not(:has-text(/)):not(:has-text(-)):not(:has(a[href*="/dl"])):remove()
> soft98.ir##[href*="SMostafa.ir"]:remove()
> ||soft98.ir/engine/modules/antibot/antibot.php^$all
> ||soft98.ir/engine/ajax/controller.php^$all
> ||file.soft98.ir^$image,redirect=click2load.html
> ||cdn.hostdl.com^$image,redirect=click2load.html
> ||linkdoni.soft98.ir^$all
> ||iranicard.ir^$all,domain=soft98.ir
> ||iranicard.ir^$removeparam=/^utm_/
> ||asiatech.ir^$all,domain=soft98.ir
> ||asiatech.ir^$removeparam=/^utm_/
> ||cdn.hostdl.com/goto/soft98^$all
> ||sysban.ir^$all,domain=soft98.ir
> ||smostafa.ir^$all,domain=soft98.ir
> @@||soft98.ir^$ghide
> ```
> 
> As you can see here, I'm constantly sending PRs and they get merged quickly: https://github.com/uBlockOrigin/uAssets/pulls?q=is%3Apr+is%3Aclosed+author%3AMasterKia

I confirm! These rules work for now. Tested with Ungoogled Chromium v.103.0.5060.134 i s uBlock Origin v.1.42.4.

https://i.postimg.cc/zBgb5N7B/SssSS.png

Here it is in the picture, if it gets blocked, the site breaks. But it remains visible anyway - it's not a problem.

**Comment by MasterKia** [comment]
@Rusenche 

Use:

<details>
    <summary>Filters</summary>

```
soft98.ir##+js(rc, c6e1y2, *, stay)
soft98.ir##[src*="file.soft98.ir"]:style(color: #437593 !important; background-color: #437593 !important; border: #437593 !important;)
soft98.ir##[src*="cdn.hostdl.com"]:style(color: white !important; background-color: white !important; border: white !important;)
soft98.ir,~forum.soft98.ir##:xpath(/html/body/div[*]/div/div[*]/div/div[1]):has(a[href*="98"]):remove()
soft98.ir##[href*="linkdoni.soft98.ir"]:remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/div/section[*]/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/section[*]/div):has(img[src*="cdn.hostdl.com"]):style(color: white !important;)
soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/header):remove()
soft98.ir##:xpath(/html/body/div[*]/div/aside/div/section/div):remove()
soft98.ir##:xpath(/html/body/div[*]/div/div/div/section):remove()
soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/span):remove()
soft98.ir##:xpath(/html/body/div[*]/div[*]/main/article[*]/div[*]):not(:has(img[src*="cdn.soft98.ir"])):not(:has-text(/)):not(:has(li)):not(:has(a[href*="/dl"])):remove()
soft98.ir##[href*="SMostafa.ir"]:remove()
||soft98.ir/engine/modules/antibot/antibot.php^$all
||soft98.ir/engine/ajax/controller.php^$all
||file.soft98.ir^$image,redirect=click2load.html
||cdn.hostdl.com^$image,redirect=click2load.html
||linkdoni.soft98.ir^$all
||iranicard.ir^$all,domain=soft98.ir
||iranicard.ir^$removeparam=/^utm_/
||asiatech.ir^$all,domain=soft98.ir
||asiatech.ir^$removeparam=/^utm_/
||cdn.hostdl.com/goto/soft98^$all
||sysban.ir^$all,domain=soft98.ir
||smostafa.ir^$all,domain=soft98.ir
@@||soft98.ir^$ghide
```
</details>

I'll try to get these merged into AdGuard as well.

**Comment by Rusenche** [comment]
@MasterKia 

https://postimg.cc/fJ4sZ4Rc

Any idea on blocking this?

**Comment by MasterKia** [comment]
Remove all previous filters and use the filters I sent [here](https://github.com/AdguardTeam/AdguardFilters/pull/126425#issuecomment-1209214477).

If that doesn't help, then disable AdGuard Base filters temporarily and see if it's still like that.

**Comment by MasterKia** [comment]
@Rusenche 

You said you're using uBlock Origin, if that's the case then remove all of my filters and just update the Quick-Fixes list from uBO settings.

The related soft98 issue on uBO issue-tracker:
`https://github.com/uBlockOrigin/uAssets/issues/14296`

**Comment by Rusenche** [comment]
> Remove all previous filters and use the filters I sent [here](https://github.com/AdguardTeam/AdguardFilters/pull/126425#issuecomment-1209214477).
> 
> If that doesn't help, then disable AdGuard Base filters temporarily and see if it's still like that.

@MasterKia

I did exactly what you wrote but the result is as I showed it in the picture.

https://postimg.cc/fJ4sZ4Rc

> You said you're using uBlock Origin, if that's the case then remove all of my filters and just update the Quick-Fixes list from uBO settings.

> The related soft98 issue on uBO issue-tracker:
> https://github.com/uBlockOrigin/uAssets/issues/14296

I did exactly what you wrote but the result is as I showed it in the picture.

https://postimg.cc/fJ4sZ4Rc

**Comment by MasterKia** [comment]
@Rusenche Okay I can reproduce using Ungoogled Chromium.

**Pull request closed without merging**